enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DefaultNamespaceEnricher.java

```diff
@@ -95,13 +99,47 @@ public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
     public void enrich(PlatformMode platformMode, KubernetesListBuilder builder) {
 
         builder.accept(new TypedVisitor<ObjectMetaBuilder>() {
+            private String getNamespaceName() {
+                String name = null;
+                if (config.getNamespace() != null && !config.getNamespace().isEmpty()) {
+                    name = config.getNamespace();
+                }
+
+                name = builder.getItems().stream()
+                        .filter(item -> Arrays.asList(NAMESPACE_KINDS).contains(item.getKind()))
+                        .findFirst().get().getMetadata().getName();
```

Comment:
`findFirst` returns an `Optional` which might contain or not the element, so if it does not contain `getMetadata` will return a NPE.

Reply:
HI Alex, https://github.com/fabric8io/fabric8-maven-plugin/pull/1614/files#diff-6e4d1897bfc7a57cabed4541720c8ab0R117 I'm doing a check for Namespace and Project Kind here, before calling this private method. So null case will never occur.

Reply:
thanks for the clarification